### DISCUSSION TITLE
"Session was used while the request was declared stateless." when i used Bref with ApiPlatform

### DISCUSSION THREAD

**Discussion by piolet** [brefphp/bref#1713] [Support]
I build an API with Aymfony and ApiPlatform. Withtout Bref, everything is ok,  i can run the symfony server without any problem.
But, when i update Kernel like in documentation (https://bref.sh/docs/symfony/getting-started#setup) and i launch `symfony serve` command, i have always this critical error :

`[Application] Jan  8 09:40:04 |CRITICA| REQUES Uncaught PHP Exception Symfony\Component\HttpKernel\Exception\UnexpectedSessionUsageException: "Session was used while the request was declared stateless." at AbstractSessionListener.php line 224 
`

i tried to change the stateless parameter in my route configuration, without success.

Is it a known problem ?

Little precision, for the moment, the problem is only on local, not during deployment or another, just local

**Comment by mnapoli** [comment]
If you are using sessions on Lambda, you might want to set them up to be stored in a central place like a database. See this documentation:

![Screen-001510](https://github.com/brefphp/bref/assets/720328/ec287435-24f8-4545-a267-35cb8d5566b1)

https://symfony.com/doc/current/session.html#store-sessions-in-a-database

Is that what you are doing already?

**Reply by piolet** [reply]
Maybe but my problem is local, not yet with lambda :/

**Reply by piolet** [reply]
I followed this instruction about cache and sessions (https://bref.sh/docs/symfony/caching)
But for me (maybe i'm wrong) this is only for lambda use case, isn't ?

Here (https://bref.sh/docs/local-development) there are no specific instructions for sessions, everything should work fine.

I'm sure I've missed something (again ;) ), but as usual, I can't work out what.

**Comment by piolet** [comment]
Hi @mnapoli,

I think I've found the root of the problem, and (maybe) the solution.

Some of routes in my API use a particular authenticator (other than one with a JWT token) and this authenticator stores the token in the session. 

A priori, by indicating, specifically for these routes, that they are not stateless, it works.

However, I don't know what the side-effects might be :/.
